### PR TITLE
Initial Version of TextDream

### PR DESCRIPTION
**Pull request by Sparkier**
This is the initial working version of TextDream. It does not include a Blog post yet.

### CODE REVIEW

**Comment by Sparkier** [comment]
@googlebot I signed it!

File: text-dream/public/index.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE html>
```
Comment (jameswex) [inline text-dream/public/index.html]: what is the favicon? where from?

File: text-dream/public/index.html [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+<html lang="en">
```
Comment (jameswex) [inline text-dream/public/index.html]: all html/js files should have copyright headers

File: text-dream/src/App.test.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
```
Comment (jameswex) [inline text-dream/src/App.test.js]: add how to run tests to readme

File: text-dream/public/index.html (hunk 1 above)
Comment (Sparkier) [reply text-dream/public/index.html]: Oh, I started this from a template I have and this favicon was from another project. I just forgot to remove it. We need to add a different one later.

File: text-dream/public/index.html (hunk 2 above)
Comment (Sparkier) [reply text-dream/public/index.html]: Done.

File: text-dream/src/App.test.js (hunk 3 above)
Comment (Sparkier) [reply text-dream/src/App.test.js]: Done, although I have not written tests yet.

File: text-dream/src/actions/index.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+import * as types from './types';
+
+/**
+ * Triggers the addition of a new dreaming element.
+ *
+ * @param {object} element - the element to be added to the visualization.
+ * @return {object} the action to be dispatched for this change.
+ */
+export function addDreamingElement(element) {
+  if (element.type === 'top_words') {
```
Comment (jameswex) [inline text-dream/src/actions/index.js]: why this special check for top_words?

File: text-dream/src/cardprocessing/index.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        params={dreamingElement.params}
+        elementIndex={elementIndex}/>;
+      break;
+    case 'reconstruct_shifted':
+      dreamingCard = <ShiftedReconstruct
+        results={dreamingElement.results}
+        params={dreamingElement.params}
+        elementIndex={elementIndex}/>;
+      break;
+    case 'top_words':
```
Comment (jameswex) [inline text-dream/src/cardprocessing/index.js]: should these strings be constants as well? seen this string in multiple files already

File: text-dream/src/components/bodies/SimilarEmbeddingsBodyComponent.js [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+        .attr('x', 0)
+        .attr(
+            'width',
+            function(d) {
+              return xScaleDistance(d.distance);
+            })
+        .attr(
+            'y',
+            function(d) {
+              return yScale(d.token)+yScale.bandwidth()/2.0;
```
Comment (jameswex) [inline text-dream/src/components/bodies/SimilarEmbeddingsBodyComponent.js]: spaces around math operators

File: text-dream/src/actions/index.js (hunk 4 above)
Comment (Sparkier) [reply text-dream/src/actions/index.js]: For top_words, i need a interation variable in the state, so I can use a slider etc. for going over iterations.
This variable needs to get initialized and its not really intuitive to require it in the input as it is only useful for the application to keep track of what the user wants to see.

File: text-dream/src/cardprocessing/index.js (hunk 5 above)
Comment (Sparkier) [reply text-dream/src/cardprocessing/index.js]: Done

File: text-dream/src/components/bodies/SimilarEmbeddingsBodyComponent.js (hunk 6 above)
Comment (Sparkier) [reply text-dream/src/components/bodies/SimilarEmbeddingsBodyComponent.js]: Done

**Pull request merged**